tests/test_function.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+import numpy as np
+
+from devito import Dimension, Function, Grid, Eq, Operator
+from devito.data.allocators import DataReference
+
+
+class TestRebuild:
+    """Tests for rebuilding of Function types."""
+
+    def test_w_new_dims(self):
+        x = Dimension('x')
+        y = Dimension('y')
+        x0 = Dimension('x0')
+        y0 = Dimension('y0')
+
+        f = Function(name='f', dimensions=(x, y), shape=(11, 11))
+
+        dims0 = (x0, y0)
+        dims1 = (x, y0)
+
+        f0 = f._rebuild(dimensions=dims0)
+        f1 = f._rebuild(dimensions=dims1)
+
+        assert f0.function is f0
+        assert f0.dimensions == dims0
+
+        assert f1.function is f1
+        assert f1.dimensions == dims1
+
+
+class TestDataReference:
```

Comment:
... while this class belongs to test_data, where other allocators are tested already